crates/sui-framework/sources/object.move

```diff
@@ -16,12 +17,18 @@ module sui::object {
     #[test_only]
     friend sui::test_scenario;
 
+    /// The recipient is not an object
+    const ENotAnObjectRecipient: u64 = 101;
+
     /// The hardcoded ID for the singleton Sui System State Object.
     const SUI_SYSTEM_STATE_OBJECT_ID: address = @0x5;
 
     /// The hardcoded ID for the singleton Clock Object.
     const SUI_CLOCK_OBJECT_ID: address = @0x6;
 
+    /// Currently unused. The recipient is an object
+    const OBJECT_RECIPIENT_KIND: u8 = 1;
```

Comment:
The pattern of distributing the kinds across modules is a little strange to me, vs keeping these constants and a `friend` constructor inside `recipient.move` -- feels quite easy to forget to keep everything up-to-date if we were ever to change things (but at the same time, I don't see us adding many new kinds here, so maybe not a big deal).

Reply:
Ah, sure I could do that. It started because I couldn't actually make an ID in that file, but let me still move the construction and such back

Reply:
I though this was a master stroke on your part to leverage the fact that private function signatures can change in upgrades -- let's just say that's what it was, and keep these private.